apps/aecore/lib/aecore/multi_node_test_framework/worker.ex

```diff
@@ -0,0 +1,611 @@
+  @spec compare_nodes_by_top_block(String.t(), String.t()) :: :synced | :not_synced
+  def compare_nodes_by_top_block(node_name1, node_name2) do
+    get_node_top_block(node_name1)
+    get_node_top_block(node_name2)
+    GenServer.call(__MODULE__, {:compare_nodes_by_top_block, node_name1, node_name2})
+  end
+
+  @doc """
+    Updates registered oracles state for 2 nodes and compares them.
+    If registered oracles are equal - they are synced
+  """
+  def compare_nodes_by_registered_oracles(node_name1, node_name2) do
+    registered_oracles(node_name1)
+    registered_oracles(node_name2)
+    GenServer.call(__MODULE__, {:compare_nodes_by_registered_oracles, node_name1, node_name2})
+  end
+
+  def compare_nodes_by_oracle_interaction_objects(node_name1, node_name2) do
+    oracle_interaction_objects(node_name1)
+    oracle_interaction_objects(node_name2)
+
+    GenServer.call(
+      __MODULE__,
+      {:compare_nodes_by_oracle_interaction_objects, node_name1, node_name2}
+    )
+  end
+
+  @doc """
+    Gets the process ports info.
+  """
+  def ports_info do
+    GenServer.call(__MODULE__, {:ports_info})
+  end
+
+  @doc """
+    Sending command in a given node.
+  """
+  @spec send_command(String.t(), String.t()) :: :ok | :unknown_node
+  def send_command(node_name, cmd) do
+    # adding \n to cmd to imitate pressing enter in iex shell
+    cmd = cmd <> "\n"
+    GenServer.call(__MODULE__, {:send_command, node_name, cmd})
+  end
+
+  @spec get_pool(String.t()) :: :ok | :unknown_node
+  def get_pool(node_name) do
+    send_command(node_name, "Pool.get_pool()")
+  end
+
+  @doc """
+    Kills the process, releases the port and removes the folder of the node
+  """
+  @spec delete_node(String.t()) :: :ok | :unknown_node
+  def delete_node(node_name) do
+    GenServer.call(__MODULE__, {:delete_node, node_name})
+  end
+
+  def delete_all_nodes do
+    GenServer.call(__MODULE__, {:delete_all_nodes}, 10000)
+  end
+
+  defp update_registered_oracles_state(node_name) do
+    GenServer.call(__MODULE__, {:update_registered_oracles_state, node_name})
+  end
+
+  defp update_oracle_interaction_objects_state(node_name) do
+    GenServer.call(__MODULE__, {:update_oracle_interaction_objects_state, node_name})
+  end
+
+  @spec update_top_block_state(String.t()) :: :ok | String.t()
+  defp update_top_block_state(node_name) do
+    GenServer.call(__MODULE__, {:update_top_block_state, node_name})
+  end
+
+  @spec update_peers_map(String.t()) :: :ok
+  defp update_peers_map(node_name) do
+    GenServer.call(__MODULE__, {:update_peers_map, node_name})
+  end
+
+  # oracles
+  @spec oracle_interaction_objects(String.t()) :: :ok | :unknown_node
+  def oracle_interaction_objects(node_name) do
+    send_command(node_name, "int_object = Chain.oracle_interaction_objects()")
+
+    # converting the keys which are binary to string
+    send_command(
+      node_name,
+      "oracle_int_obj_for_encoding = for {k,v} <- int_object, into: %{}, do: {Base.encode32(k), v}"
```

Comment:
Wouldn't it be actually easier to use `Serialization.rlp_encode(oracle, :oracle_query)` for serialization? (This will also be more resistant to changes in nodes) Or is there some reason that we can't serialize using existing serialization?

Reply:
Or maybe we should use `:erlang.term_to_binary` as this will guarantee everything is always encoded and decoded properly (and we don't need any compatibility in tests with non elixir nodes)?